src/payment/strategies/square/square-form.ts

```diff
@@ -39,7 +93,16 @@ export interface SquareFormElement {
 export interface SquareFormCallbacks {
     paymentFormLoaded?(form: SquarePaymentForm): void;
     unsupportedBrowserDetected?(): void;
-    cardNonceResponseReceived?(errors: any, nonce: string): void;
+    cardNonceResponseReceived?(
+        errors?: NonceGenerationError[],
+        nonce?: string,
+        cardData?: CardData,
+        billingContact?: Contact,
+        shippingContact?: Contact): void;
+    methodsSupported?(methods: any): void;
+    createPaymentRequest?(): void;
+    validateShippingContact?(errors: SquareValidationErrors): void;
```

Comment:
⚠️⚠️⚠️⚠️
 
I feel like you might have misinterpreted the documentation here.

https://docs.connect.squareup.com/payments/sqpaymentform/overview#the-validateshippingcontact-callback

I think this works the other way around, they give u a shipping address and you validate that it is valid, and if it is not, you have to return the object with the errors.

```
    /*
     * callback function: validateShippingContact
     * Triggered when: a shipping address is selected/changed in a digital
     *                 wallet UI that supports address selection.
     */
    validateShippingContact: function (contact) {

      var validationErrorObj ;
      /* ADD CODE TO SET validationErrorObj IF ERRORS ARE FOUND */
      return validationErrorObj ;
    },
```

Reply:
I'll remove that callback then.

Reply:
Could you double check you have the other callbacks right?

Reply:
👍 Absolutelly